plugins/chat/assets/javascripts/discourse/components/chat-message.js

```diff
@@ -86,6 +96,25 @@ export default class ChatMessage extends Component {
   @action
   teardownChatMessage() {
     cancel(this._invitationSentTimer);
+
+    this.args.message.mentionedUsers.forEach((user) => {
+      user.stopTrackingStatus();
+      user.off("status-changed", this, "refreshStatusOnMentions");
+    });
+  }
+
+  @action
```

Comment:
can we use the decorators hooks for this? each time a message is updated the version will be incremented and decorates hooks will be replayed, look at what we do for dates for example

Reply:
This is tricky. I initially implemented it as a decorator in `initializers/chat-decorators.js`, then I felt that it's better to reimplement it this way.

Note that we need to re-render status in the next cases:
1. when a message is posted
2. when a message is edited
3. when we receive a message bus message with status update for a mentioned user

When I implemented it as a decorator, I needed to redecorate the message in the third case, which calls all decorators, while in fact there is no need to re-render everything, it's enough to only update user status.

By the way, as a side note, another thing that bothers me is that these decorators are implemented using an initializer. That couples the chat-message component with the application startup logic. You can notice that if you try to write a component test for the `chat-message` component, none of the decorators will work by default. Of course, we can initialize decorators before running a component test, but in general, the fact that we can't just test a component independently is a sign that that logic actually naturally belongs to the component itself. I think it's something we can think about, looks like it's possible to reimplement these decorators without using initializer, that may be simpler.

So maybe let's better to leave re-rendering using status in the component as it's implemented now? It's more efficient, easier to understand and easier to test?